wm-system/src/main/kotlin/com/ajaxproject/warehouse/service/grpc/DeleteProductGrpcService.kt

```diff
@@ -0,0 +1,28 @@
+package com.ajaxproject.warehouse.service.grpc
+
+import com.ajaxproject.api.internal.warehousesvc.input.reqreply.product.DeleteProductByIdRequest
+import com.ajaxproject.api.internal.warehousesvc.input.reqreply.product.DeleteProductByIdResponse
+import com.ajaxproject.warehouse.service.ProductService
+import org.springframework.stereotype.Service
+import reactor.kotlin.core.publisher.toMono
+
+@Service
+class DeleteProductGrpcService(
+    private val productService: ProductService
+) {
```

Comment:
1) Are you sure you need to Wrap Product Service?
Buildings responses is totally Controller's job.
2) Are this Service reusable or he is just work with Grpc controller?
3) What logic this service do? In fact he is just add buildResponse - which is not Service job
4) Your @GrpcService class is like @RestController - he should receiving request, invoking Service and builds response. I don't see why @GrpcService delivers part of his job to Some Service

Reply:
Good points.
I agree that it's far from ideal. But let's leave it as is. We haven't so many time to make the design brilliant.